core/modules/comment/comment.module

```diff
@@ -2336,6 +2357,87 @@ function comment_file_download_access($field, $entity_type, $entity) {
   }
 }
 
+/**
+ * Implements hook_cron().
+ */
+function comment_cron() {
+  _comment_close_all();
+}
+
+/**
+ * Close all comments that meet the current threshold.
+ *
+ * @param $now
+ *  A UNIX timestamp for the expiration time.
+ */
+function _comment_close_all($now = NULL) {
+  if (is_null($now)) {
+    $now = REQUEST_TIME;
+  }
+  $content_types = node_type_get_names();
+
+  foreach ($content_types as $type => $name) {
+    // Set it up.
+    $config = config('node.type.' . $type);
+    $days = $config->get('settings.comment_close_days');
+    $enable = $config->get('settings.comment_close_enable');
+
+    // If a day is set, do this one.
+    if (!empty($enable) && !empty($days)) {
+      $oldest_allowed = _comment_closer_oldest_allowed($days, $now);
+
+      // Fetch nodes that should have comments closed.
+      $result = db_query("SELECT nid FROM {node} WHERE created < :oldest AND type = :type AND comment = :comment",
+        array(':oldest' => $oldest_allowed, ':type' => $type, ':comment' => COMMENT_NODE_OPEN)
+      );
+      $nids = $result->fetchCol();
+
+      // Update those rows.
+      foreach($nids as $nid) {
+        $node = node_load($nid);
+        $node->comment = COMMENT_NODE_CLOSED;
+        node_save($node);
+      }
+
+      $count  = $result->rowCount();
+      if ($count) {
+        $vars = array(
+          '!count' => $count,
+          '@type' => $name,
+          '!date' => format_date($oldest_allowed),
+        );
+        $msg = 'Closed comments on !count @type posts created at, or before, !date.';
+        watchdog('comment', $msg, $vars, WATCHDOG_NOTICE);
+      }
+    }
+  }
+}
+
+/**
+ * Helper function for generating date from age interval.
+ */
+function _comment_closer_oldest_allowed($days, $now = NULL) {
```

Comment:
If this function is only used once, perhaps consolidate into `_comment_close_all`?